vpn/tls.go

```diff
@@ -88,9 +92,34 @@ type handshaker interface {
 
 // defaultTLSFactory returns an implementer of the handshaker interface; that
 // is, the default tls.Client factory.
-func defaultTLSFactory(conn net.Conn, config *tls.Config) handshaker {
-	return tls.Client(conn, config)
+// we're not using the default factory right now, but it comes handy to be able
+// to compare the fingerprints with a golang TLS handshake.
+func defaultTLSFactory(conn net.Conn, config *otls.Config) handshaker {
+	return otls.Client(conn, config)
+}
+
+// vpnClientHelloHex is the hexadecimal respresentation of a capture from the reference openvpn implementation.
+const vpnClientHelloHex = `16030100e8010000e40303fe0b6526568ada469f8a7996b79b2598208481dc43fe56081614c7e0e8b9bd8920ddb7565358d398109fb7934c077eb0234c98839b2578046904849b2b76156ab1000a130213031301c02c00ff01000091000b000403000102000a00160014001d0017001e00190018010001010102010301040016000000170000000d002a0028040305030603080708080809080a080b080408050806040105010601030303010302040205020602002b00050403040303002d00020101003300260024001d0020a9cf1d61f8caee159b9a4dc684d9319e2349f80f6e82ff7b755b820ff33fa75f`
+
+// parrotTLSFactory returns an implementer of the handshaker interface; in this
+// case, a parroting implementation.
+func parrotTLSFactory(conn net.Conn, config *tls.Config) handshaker {
+	fingerprinter := &tls.Fingerprinter{AllowBluntMimicry: true}
+
+	rawOpenVPNClientHelloBytes, err := hex.DecodeString(vpnClientHelloHex)
+	if err != nil {
+		logger.Error(err.Error())
```

Comment:
I would argue that here we should either return an error or panic (but maybe panicking is a bit too aggressive and we should return an error, especially considering this is a library not just a tool)